src/components/snapshot/SnapshotPopup.tsx

```diff
@@ -0,0 +1,247 @@
+import { Link } from 'react-router-dom';
+import clsx from 'clsx';
+import { openSnapshotDialog, getSnapshotById, getSnapshotPolicy } from 'actions/index';
+import { CustomTheme } from '@mui/material/styles';
+import { Action } from 'redux';
+import { TdrState } from 'reducers';
+import { PolicyModel, SnapshotModel } from 'generated/tdr';
+import { SnapshotIncludeOptions } from '../../constants';
+
+const styles = (theme: CustomTheme) =>
+  createStyles({
+    snapshotName: {
+      backgroundColor: theme.palette.primary.light,
+      borderRadius: '4px 4px 0px 0px',
+    },
+    content: {
+      padding: theme.spacing(2),
+    },
+    withIcon: {
+      display: 'flex',
+      alignItems: 'center',
+    },
+    inline: {
+      marginRight: theme.spacing(1),
+    },
+    actions: {
+      paddingTop: theme.spacing(2),
+      paddingBottom: theme.spacing(0.5),
+      textAlign: 'end',
+    },
+    listItem: {
+      listStyleType: 'none',
+      marginBottom: theme.spacing(1),
+    },
+    bodyText: {
+      paddingBottom: theme.spacing(2),
+    },
+    light: {
+      color: 'rgba(0, 0, 0, 0.54)',
+    },
+    chip: {
+      backgroundColor: theme.palette.primary.light,
+      margin: theme.spacing(0.5),
+      marginLeft: '0px',
+    },
+    centered: {
+      textAlign: 'center',
+    },
+    exitButton: {
+      position: 'absolute',
+      top: 0,
+      right: 0,
+      padding: '1em',
+      width: '3em',
+      height: '3em',
+      cursor: 'pointer',
+      ...theme.mixins.jadeLink,
+    },
+    title: {
+      marginTop: '10px',
+      marginRight: '16px',
+      paddingBottom: '15px',
+    },
+    jadeLink: {
+      ...theme.mixins.jadeLink,
+    },
+  });
+
+interface SnapshotPopupProps {
+  classes: ClassNameMap;
+  dispatch: Dispatch<Action>;
+  filterData: any;
+  isOpen: boolean;
+  policies: Array<PolicyModel>;
+  snapshot: SnapshotModel;
+}
+
+function SnapshotPopup({
+  classes,
+  dispatch,
+  filterData,
+  isOpen,
+  policies,
+  snapshot,
+}: SnapshotPopupProps) {
+  /**
+   * When a snapshot gets created, the empty snapshot prop gets populated with a brief summary.
+   * We want to use that snapshot id to get the full object and the policies for the dialog.
+   */
+  const prevSnapshotRef = useRef<SnapshotModel>(snapshot);
+  useEffect(() => {
+    if (prevSnapshotRef.current && _.isEmpty(prevSnapshotRef.current) && !_.isEmpty(snapshot)) {
+      prevSnapshotRef.current = snapshot;
+      dispatch(
+        getSnapshotById({
+          snapshotId: snapshot.id,
+          include: [
+            SnapshotIncludeOptions.SOURCES,
+            SnapshotIncludeOptions.TABLES,
+            SnapshotIncludeOptions.RELATIONSHIPS,
+            SnapshotIncludeOptions.ACCESS_INFORMATION,
+            SnapshotIncludeOptions.PROFILE,
+            SnapshotIncludeOptions.DATA_PROJECT,
+          ],
+        }),
+      );
+      dispatch(getSnapshotPolicy(snapshot.id));
+    }
+  }, [snapshot, dispatch]);
```

Comment:
Do we want to subscribe on `dispatch` here? I know we ran into some issues with that elsewhere